homeassistant/components/screenlogic/__init__.py

```diff
@@ -159,11 +159,9 @@ async def _async_update_data(self):
         """Fetch data from the Screenlogic gateway."""
         try:
             await self._async_update_configured_data()
-        except ScreenLogicError as error:
-            _LOGGER.warning("Update error - attempting reconnect: %s", error)
+        except (ScreenLogicError, ScreenLogicWarning) as ex:
+            _LOGGER.warning("Update error - attempting reconnect: %s", ex)
             await self._async_reconnect_update_data()
```

Comment:
Does this need to be exposed to HomeAssistant?  Can the lib decide on if it needs to reconnect or not?

Reply:
It can. In fact the most recent release (v0.8.0) includes retrying failed requests, but the logs provided for this issue describe a networking failure mode I didn't anticipate, so it really needs more granular error handling which is going to take a little bit of time.
I wanted to unblock users so I made this simple fix while the more extensive api solution is worked on.

Reply:
The judging of when it should stop trying and raise is the more granular part. I'd like to find a balance to between retrying within the api, and a user's command hanging waiting for confirmation of failure.

Reply:
Let's do this for now in a patch release and handle it in the library in the future.

Reply:
Sounds good. I am looking into it, but at the moment it's looking like it might be a bit of a refactor.